sqlmesh/cli/main.py

```diff
@@ -94,21 +99,28 @@ def cli(
 
 
 @cli.command("init")
+@click.option(
+    "dialect",
+    required=True,
+    type=str,
+    help=f"Default model dialect. Supported values: {_sqlglot_dialects()}.",
+)
 @click.option(
     "-t",
     "--template",
     type=str,
-    help="Project template. Support values: airflow, dbt, default.",
+    help="Project template. Supported values: airflow, dbt, default.",
 )
 @click.pass_context
 @error_handler
-def init(ctx: click.Context, template: t.Optional[str] = None) -> None:
+def init(ctx: click.Context, dialect: str, template: t.Optional[str] = None) -> None:
     """Create a new SQLMesh repository."""
     try:
         project_template = ProjectTemplate(template.lower() if template else "default")
     except ValueError:
         raise click.ClickException(f"Invalid project template '{template}'")
-    init_example_project(ctx.obj, template=project_template)
+    _ = Dialects(dialect.lower())
```

Comment:
what is this? why do you have _ ?

Reply:
you don't need to do it at all i think, doesn't it already crash when you pass an invalid one in?

Reply:
Are you able to use click's choice option?
> https://click.palletsprojects.com/en/8.1.x/options/#choice-options

Reply:
yes, the checks i added are redundant - i deleted them all